pkg/sql/colexec/table_scan/table_scan.go

```diff
@@ -90,7 +88,7 @@ func (tableScan *TableScan) Call(proc *process.Process) (vm.CallResult, error) {
 	}
 
 	if tableScan.ctr.buf != nil {
-		proc.PutBatch(tableScan.ctr.buf)
+		tableScan.ctr.buf.Clean(proc.GetMPool())
```

Comment:
Later, result.batch uses buf.   And eval, if proj list is not null.    This seems all fine, except this buf cannot be cleaned here.    It is just wrong.